tests/rest/test_samples_entity_search.py

```diff
@@ -0,0 +1,101 @@
+
+    def test_entity_search_response_is_json(self):
+        """Test that Entity Search API returns responses in JSON format"""
+        response = entity_search_basic(
+            "Abu Hayyan",
+            subscription_key=self.env.get(
+                "BING_SEARCH_V7_ENTITY_SEARCH_SUBSCRIPTION_KEY"
+            ),
+        )
+        try:
+            response.json()
+        except JSONDecodeError:
+            self.fail(
+                "Failed to decode response as JSON, check if valid JSON is returned"
+            )
+
+    def test_entity_search_no_auth(self):
+        """Test that Entity Search API returns 401 if authorization fails"""
+        with self.assertRaises(HTTPError):
+            entity_search_basic("Egypt", subscription_key="")
+
+    def test_entity_search_response_object_type(self):
+        """Test that Entity Search API returns the correct type hint"""
+        response = entity_search_basic(
+            "Ghiyath al-Din Muhammad",
+            subscription_key=self.env.get(
+                "BING_SEARCH_V7_ENTITY_SEARCH_SUBSCRIPTION_KEY"
+            ),
+        )
+        try:
+            self.assertEqual(response.json()["_type"], "SearchResponse")
+        except KeyError:
+            self.fail("The response object doesn't include the type hint")
+
+    def test_entity_search_response_object_structure(self):
+        """Test that Entity Search API responses follow the correct structure"""
+        response = entity_search_basic(
+            "Arab",
+            subscription_key=self.env.get(
+                "BING_SEARCH_V7_ENTITY_SEARCH_SUBSCRIPTION_KEY"
+            ),
+        )
+        response_json = response.json()
+        self.assertIn("entities", response_json)
+        self.assertIn("rankingResponse", response_json)
+        try:
+            self.assertGreater(len(response_json["entities"]["value"]), 0)
+        except KeyError:
+            self.fail("The response object doesn't include any entity results")
+
+    # This fails because of a shortcoming in the API itself not because of the sample code
+    def test_entity_search_required_parameter_query(self):
+        """Test that Entity Search API returns an error if required parameter is empty"""
+        with self.assertRaises(HTTPError):
+            entity_search_basic(
+                subscription_key=self.env.get(
+                    "BING_SEARCH_V7_ENTITY_SEARCH_SUBSCRIPTION_KEY"
+                ),
+                query="",
+            )
+
+    # TODO: This currently fails because HTTPError is thrown and the 'response' variable is not populated
+    # def test_entity_search_error_response_object_structure(self):
+    #     """Test the structure of the Error Response"""
+    #     response = entity_search_basic("", "")
+    #     try:
+    #         self.assertEqual(response.json()["_type"], "ErrorResponse")
+    #     except KeyError:
+    #         self.fail("The response object is not of type 'ErrorResponse'")
```

Comment:
I'm not sure I got it but if you expect the unit test to fail, you can also handle it as a successful run not failing one. [reference ](https://www.geeksforgeeks.org/test-if-a-function-throws-an-exception-in-python/)

Reply:
done